lib/saluki-components/src/transforms/name_filter/mod.rs

```diff
@@ -0,0 +1,238 @@
+use std::ops::Deref;
+
+use async_trait::async_trait;
+use memory_accounting::{MemoryBounds, MemoryBoundsBuilder};
+use saluki_config::GenericConfiguration;
+use saluki_core::{
+    components::{
+        transforms::{Transform, TransformBuilder, TransformContext},
+        ComponentContext,
+    },
+    topology::OutputDefinition,
+};
+use saluki_error::GenericError;
+use saluki_event::{metric::Metric, DataType};
+use serde::Deserialize;
+use tokio::select;
+use tracing::{debug, error};
+
+#[derive(Deserialize)]
+/// Name Transform.
+pub struct NameFilterConfiguration {
+    #[serde(default, rename = "statsd_metric_namespace")]
+    metric_prefix: String,
+
+    #[serde(
```

Comment:
I think you meant for this to be `serde`?

Reply:
Sorry, bit confused by this 😅

Reply:
Weird, when I reviewed this, it was literally `#[ core (` instead of `#[serde(`. 😮